### ISSUE TITLE
Deprecate track1 data plane container registry

### ISSUE THREAD

**Issue by pallavit** [Azure/azure-sdk-for-net#31979]
track1 data plane SDK Microsoft.Azure.ContainerRegistry is not yet deprecated. We need to do the following to deprecate it.

- [ ] Add migration guide,
- [ ] Analyze the feature set supported by track1 and ensure they are either supported on track2 or are not needed.
- [ ] Mark the package deprecated on nuget.

**Comment by annelo-msft** [comment]
Update: @hector-norza is investigating whether the requirements apply to preview libraries, since [Microsoft.Azure.ContainerRegistry](https://www.nuget.org/packages/Microsoft.Azure.ContainerRegistry/1.0.0-preview.2) never GA'ed.

**Comment by pallavit** [comment]
Adding @mario-guerra. Once we GA the track2 .NET ACR SDK (May 2023) we should be able to deprecate track1 SDK. Could you please help us with the deprecation here.

**Comment by mario-guerra** [comment]
@pallavit yes, I can definitely take care of deprecating the old package once the Track 2 library is GA.
